src/lambda/local/debugConfiguration.ts

```diff
@@ -3,11 +3,27 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import * as path from 'path'
+import * as vscode from 'vscode'
 import { nodeJsRuntimes } from '../../lambda/models/samLambdaRuntime'
 import { SamLaunchRequestArgs } from '../../shared/sam/debugger/samDebugSession'
 import { dotNetRuntimes, pythonRuntimes, RuntimeFamily } from '../models/samLambdaRuntime'
+import {
+    CodeTargetProperties,
+    TemplateTargetProperties,
+    AwsSamDebuggerConfiguration,
+} from '../../shared/sam/debugger/awsSamDebugConfiguration'
+import { tryGetAbsolutePath } from '../../shared/utilities/workspaceUtils'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { CloudFormation } from '../../shared/cloudformation/cloudformation'
+import { CloudFormationTemplateRegistry } from '../../shared/cloudformation/templateRegistry'
 
 export const DOTNET_CORE_DEBUGGER_PATH = '/tmp/lambci_debug_files/vsdbg'
+export const AWS_SAM_DEBUG_TYPE = 'aws-sam'
+export const DIRECT_INVOKE_TYPE = 'direct-invoke'
+export const TEMPLATE_TARGET_TYPE: 'template' = 'template'
+export const CODE_TARGET_TYPE: 'code' = 'code'
+export const AWS_SAM_DEBUG_TARGET_TYPES = [TEMPLATE_TARGET_TYPE, CODE_TARGET_TYPE]
```

Comment:
these symbols (and functions below) were moved from `src/shared/sam/debugger/awsSamDebugger.ts` to avoid circular dependencies. 

Really, some of these files should be merged together: it is not clear how `debugConfiguration.ts` differs semantically from `awsSamDebugger.ts` ... (similar comment for numerous other gratuitous demarcations in the codebase)

Reply:
`awsSamDebugger.ts` directly relates to the `aws-sam` `vscode.DebugConfigurationProvider` whereas `debugConfiguration.ts` should be tied to running the launch config generated from `awsSamDebugger.ts`.  If anything, I'd suggest renaming `awsSamDebugger.ts` to something a bit more descriptive (`awsSamDebugConfigProvider.ts`?)

Reply:
There is plenty of shuffling to do, but it's out of scope for this PR.

Reply:
FYI looks like a fair amount refactoring to `awsSamDebugger` was done in https://github.com/aws/aws-toolkit-vscode/pull/972 . Should make these distinctions more clear (https://github.com/aws/aws-toolkit-vscode/pull/971 will be refactored into those files).

Reply:
yes I'm in the middle of resolving that divergence from #972